### DISCUSSION TITLE
Add callback function to use-clipboard options

### DISCUSSION THREAD

**Discussion by kimfucious** [mantinedev/mantine#6650] [Feature requests]
It would be nice to have the ability to execute a function when the timeout of the `use-clipboard` has been met.

Here's a use case:

- There is a list of items and each item has an action icon that uses `clipboard.copy()`.
- The action icon is changes color based on `clipboard.copied`.

In this scenario, all icons will change color, so a workaround can be implemented to set the copied item in local state:

```ts
const [copiedThing, setCopiedThing] = useState("");
```
```ts
onClick={() => {
    clipboard.copy(item.id);
    setCopiedThing(item.id);
}}
```

This way, the conditional coloring can be more granular, using both `clipboard.copied` and `copiedThing`.

The callback function that I am suggestion could be executed after the timeout to run `setCopiedThing("")`.

At the moment, I'm writing and calling the callback, using setTimeout.

This is not difficult, but I imagine there would be other scenarios when a having a "built-in" callback would save some time and effort.

```ts
<ActionIcon
    color={
        clipboard.copied && copiedRecipeId === item.id
            ? "green"
            : "red"
    }
    onClick={() => {
        clipboard.copy(item.id);
        setCopiedRecipeId(item.id);
        setTimeout(() => {
            setCopiedRecipeId("");
        }, CLIPBOARD_TIMEOUT);
    }}
    radius={"xl"}
    variant="subtle"
>
    <BsClipboard2Plus />
</ActionIcon>
```

**Comment by rtivital** [comment]
Use CopyButton to avoid this issue entirely – https://mantine.dev/core/copy-button/

**Comment by kimfucious** [comment]
@rtivital Awesome, thank you!

I just implemented this, and it works great.

I wasn't aware that CopyButton existed.  You're adding so many awesome features to Mantine, it's hard to keep up with all of them 😄
